osu.Desktop/GameStateBroadcaster.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Concurrent;
+using System.Net;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Threading;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets;
+
+namespace osu.Desktop
+{
+    public class GameStateBroadcaster : Component
+    {
+        private readonly ConcurrentBag<WebSocketClient> clients = new ConcurrentBag<WebSocketClient>();
+
+        private readonly HttpListener listener = new HttpListener();
+
+        private readonly CancellationTokenSource cancellationToken = new CancellationTokenSource();
+
+        private readonly Bindable<RulesetInfo> ruleset = new Bindable<RulesetInfo>();
+
+        private readonly Bindable<WorkingBeatmap> beatmap = new Bindable<WorkingBeatmap>();
+
+        private ScheduledDelegate broadcastSchedule;
+
+        public GameStateBroadcaster()
+        {
+            listener.Prefixes.Add("http://localhost:8080/");
```

Comment:
We obviously can't use such a common port.

Reply:
Would it be fine if this can be replaced in settings?

Reply:
No, it should be well defined and hopefully follow existing standards for local port assignments (ensuring we don't collide with other apps). Will need to do some reading and provide references to how this was decided.

Reply:
In the meantime I've replaced the port number to `7270`.